hw/ip/otbn/dv/rig/rig/gens/small_val.py

```diff
@@ -61,8 +61,21 @@ def gen(self,
         assert grd_val is not None
 
         # Pick a target value. For now, we take arbitrary values in the range
-        # [-5, 5].
-        tgt_val = random.randint(-5, 5)
+        # [-64, 64]. Ideally ten percent of the time we want to have a random
+        # value that is actually divisible by 32. That way we would have valid
+        # registers that have values to point as WDR addresses for BN.XID
+        # Other values are for helping with the misaligned address calculations
+        temp_val = random.randint(-64, 64)
+        gen_for_bn = random.randint(0, 9)
+
+        if (gen_for_bn == 0) and (temp_val % 32):
+            tgt_val = temp_val << 5
+        elif (gen_for_bn == 1) and not (temp_val % 2):
+            tgt_val = temp_val + 1
+        elif (gen_for_bn == 2) and not (temp_val % 4):
+            tgt_val = temp_val - 1
+        else:
+            tgt_val = temp_val
```

Comment:
I'm not sure that the `gen_for_bn == 2` case works when `temp_val` is 3? Would it be easier just to compute a left shift amount as a function of `gen_for_bn` and then shift `temp_val` unconditionally? Maybe something like this?
```suggestion
        shifts = {0: 5, 1: 1, 2: 2}
        tgt_val = temp_val << shifts.get(gen_for_bn, 0)
```